### ISSUE TITLE
Testing of features deprecated in 1.2

### ISSUE THREAD

**Issue by kpet** [KhronosGroup/OpenCL-CTS#458]
The following features that were deprecated in 1.2 aren't covered by the CTS, they should:
- [ ] `clEnqueueBarrier`
- [ ] `clEnqueueWaitForEvents`
- [ ] `clCreateImage2D/3D` (present but guarded by a build flag)
- [ ] `clUnloadCompiler`
- [ ] `clGetExtensionFunctionAddress`

**Comment by jlewis-austin** [comment]
@kpet just to clarify, this coverage is required for 2.0+ implementations as well as 1.2?

**Comment by kpet** [comment]
Yes, required for all the versions that support these functions. None of them have been removed.

**Comment by jlewis-austin** [comment]
Great, thanks! I notice several queries are deprecated in 2.0 (e.g. CL_DEVICE_QUEUE_PROPERTIES becomes CL_DEVICE_QUEUE_ON_HOST_PROPERTIES). I assume they would also be meant to remain functional with the old names going forward?
